src/main/java/org/jahia/modules/jahiaoauth/impl/JahiaOAuthServiceImpl.java

```diff
@@ -66,6 +80,71 @@ public JahiaOAuthServiceImpl(Map<String, DefaultApi20> oAuthDefaultApi20Map) {
         }
     }
 
+    public void setSettingsService(SettingsService settingsService) {
+        this.settingsService = settingsService;
+    }
+
+    public void setConfigurationAdmin(ConfigurationAdmin configurationAdmin) {
+        this.configurationAdmin = configurationAdmin;
+    }
+
+    public void init() {
+        BundleContext bundleContext = FrameworkService.getBundleContext();
+        //Allows to configure connector without having to go on the settings UI
+        serviceTracker = new ServiceTracker<>(bundleContext, "org.jahia.modules.jahiaoauth.service.OAuthConnectorService",
+                new ServiceTrackerCustomizer<Object, Object>() {
+                    @Override
+                    public OAuthConnectorService addingService(ServiceReference serviceReference) {
+                        OAuthConnectorService oAuthConnectorService = (OAuthConnectorService) bundleContext.getService(serviceReference);
+                        loadConnector(oAuthConnectorService,
+                                serviceReference.getProperty(JahiaAuthConstants.CONNECTOR_SERVICE_NAME).toString());
+
+                        return oAuthConnectorService;
+                    }
+
+                    @Override
+                    public void modifiedService(ServiceReference serviceReference, Object o) {
+                        OAuthConnectorService oAuthConnectorService = (OAuthConnectorService) bundleContext.getService(serviceReference);
+                        loadConnector(oAuthConnectorService,
+                                serviceReference.getProperty(JahiaAuthConstants.CONNECTOR_SERVICE_NAME).toString());
+                    }
+
+                    @Override
+                    public void removedService(ServiceReference serviceReference, Object o) {
+                    }
+                });
+        serviceTracker.open();
+    }
+
+    public void destroy() {
+        serviceTracker.close();
+    }
+
+    public void loadConnector(OAuthConnectorService oAuthConnectorService, String connectorName) {
+        Configuration[] configurations = null;
+        try {
+            configurations = configurationAdmin.listConfigurations("(service.factoryPid=org.jahia.modules.auth)");
+        } catch (IOException e) {
+            logger.error("IOException reading org.jahia.modules.auth configs", e);
+        } catch (InvalidSyntaxException e) {
+            logger.error("InvalidSyntaxException reading org.jahia.modules.auth configs", e);
+        }
+
+        if (configurations != null) {
```

Comment:
I would rather prefer adding a new method to the SettingsService to be able to retrieve what you are looking for here, it already contains all the settings, so a function to expose the required data would make more sense.
A service exist to manage those settings, we should use it and modify it if necessary 

(for sure it imply modification in jahia-authentication module, but that is okay the required version is already set on jahia-authentication 1.7.0-SNAPSHOT )